### PR TITLE
Add `MixHopConv` and an example on `Cora`

### PR DESCRIPTION
**Pull request by wangbingnan136**

### CODE REVIEW

**Review by rusty1s — COMMENTED** [review]
Thanks for the PR. Is it possible for you to convert the notebook into a Python script to ease reviewing?

**Comment by wangbingnan136** [comment]
> Thanks for the PR. Is it possible for you to convert the notebook into a Python script to ease reviewing?

Ok,It is done~

File: torch_geometric/nn/conv/mixhopconv.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        return final
+
+    def message(self, x_j, norm):
+
+        return norm.view(-1, 1) * x_j
+
+
+
+
+class MixHop(nn.Module):
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/conv/mixhopconv.py:81]: I think we can just provide the basic `MixHopConv` implementation only and implement the `MixHop` model in the example.

File: examples/mixhop-pytorch-geometric-cora.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+import copy
+import torch
+import torch.optim as optim
+import torch.nn as nn
+import numpy as np
+import random
+import dgl
+import dgl.function as fn
+
+from dgl.data import CoraGraphDataset, CiteseerGraphDataset, PubmedGraphDataset,PPIDataset
```
Comment (EdisonLeeeee) [inline examples/mixhop-pytorch-geometric-cora.py:18]: Use datasets in PyG instead.

File: torch_geometric/nn/conv/mixhopconv.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            final = self.bn(final)
+        
+        if self.activation is not None:
+            final = self.activation(final)
+        
+        final = self.dropout(final)
+
+        return final
+
+    def message(self, x_j, norm):
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/conv/mixhopconv.py:74]: type hints

File: torch_geometric/nn/conv/mixhopconv.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        self.max_j =  max(self.p) + 1
+
+
+
+
+    def forward(self, x, edge_index):
+
+        outputs = []
+
+        row, col = edge_index
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/conv/mixhopconv.py:44]: Use `torch_geometric.nn.conv.gcn_conv.gcn_norm` instead.

**Comment by rusty1s** [comment]
Added via https://github.com/pyg-team/pytorch_geometric/pull/8025.

**Pull request closed without merging**